spec/requests/admin/robots_txt_controller_spec.rb

```diff
@@ -101,11 +101,17 @@
 
       it "resets robots.txt file to the default version" do
         SiteSetting.overridden_robots_txt = "overridden_content"
-        delete "/admin/customize/robots.json"
+        SiteSetting.allowed_crawler_user_agents = "test-user-agent-154"
+
+        delete "/admin/customize/robots.json", headers: { "X-Requested-With": "XMLHttpRequest" }
```

Comment:
I couldn't reproduce the problem in tests until I added this header. For some reasons, if the request isn't XHR, `render_to_string` renders the template without the application layout, but if the request is XHR, then the method renders the template with the application layout. It seems like it should be the other way around, but 🤷‍♂️

Reply:
I think settings `X-Requested-With` might also automatically set `Accept` header? So using `Accept: "text/html"` might have the same effect:

```suggestion
        delete "/admin/customize/robots.json", headers: { "Accept": "text/html" }
```

Or instead you could use:

```suggestion
        delete "/admin/customize/robots.json", xhr: true
```

Reply:
> if the request isn't XHR, `render_to_string` renders the template without the application layout, but if the request is XHR, then the method renders the template with the application layout

Oh my god I got my wires crossed🤦It's actually the opposite; if the request is XHR then you don't get the template layout which makes sense.